src/stores/provider.ts

```diff
@@ -0,0 +1,166 @@
+import { capital } from 'case'
+import { defineStore } from 'pinia'
+import { Ref, ssrRef } from '@nuxtjs/composition-api'
+import axios from 'axios'
+
+import { env } from '~/utils/env'
+import {
+  AUDIO,
+  IMAGE,
+  SupportedMediaType,
+  supportedMediaTypes,
+} from '~/constants/media'
+import { warn } from '~/utils/console'
+import { providerServices } from '~/data/media-provider-service'
+import type { MediaProvider } from '~/models/media-provider'
+import {
+  FetchState,
+  initialFetchState,
+  updateFetchState,
+} from '~/composables/use-fetch-state'
+import { useSearchStore } from '~/stores/search'
+
+export interface ProviderState {
+  providers: {
+    audio: MediaProvider[]
+    image: MediaProvider[]
+  }
+  fetchState: {
+    audio: FetchState
+    image: FetchState
+  }
+}
+
+export const initialProviderState: ProviderState = Object.freeze({
+  providers: {
+    [AUDIO]: [],
+    [IMAGE]: [],
+  },
+  fetchState: {
+    [AUDIO]: { ...initialFetchState },
+    [IMAGE]: { ...initialFetchState },
+  },
+})
+
+/**
+ * Sorts providers by their source_name property.
+ * @param data - initial unordered list of providers
+ */
+const sortProviders = (data: MediaProvider[]): MediaProvider[] => {
+  return [...data].sort((sourceObjectA, sourceObjectB) => {
+    const nameA = sourceObjectA.source_name.toUpperCase()
+    const nameB = sourceObjectB.source_name.toUpperCase()
+    return nameA.localeCompare(nameB)
+  })
+}
+/**
+ * Timestamp is used to limit the update frequency to one every 60 minutes per request.
+ */
+const lastUpdated: Ref<Date | null> = ssrRef(null)
+
+const updateFrequency = parseInt(env.providerUpdateFrequency, 10)
+
+export const useProviderStore = defineStore('provider', {
+  state: (): ProviderState => ({
+    providers: {
+      [AUDIO]: [],
+      [IMAGE]: [],
+    },
+    fetchState: {
+      [AUDIO]: { ...initialFetchState },
+      [IMAGE]: { ...initialFetchState },
+    },
+  }),
```

Comment:
This can use the `initialProviderState` object, right? Just need to `deepclone` it if it's frozen.

Reply:
I removed the `initialProviderState` object altogether, because it was overcomplicating things (deepFreeze/deepclone), and not really serving much purpose.